python_backend/database/storage.py

```
@@ -209,4 +210,172 @@ def save_negative_preferences(self, user_id: str, preferences: Dict[str, Any]) -
+                    generated_answer=query_data.get('generated_answer'),
+                    answer_quality_score=query_data.get('answer_quality_score'),
+                    model_used=query_data.get('model_used', 'gpt-4'),
+                    total_generation_time_ms=query_data.get('total_generation_time_ms', 0),
+                    user_rating=query_data.get('user_rating'),
+                    user_feedback=query_data.get('user_feedback'),
+                    was_helpful=query_data.get('was_helpful')
+                )
+
+                db.add(rag_query)
+                db.commit()
+                return True
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)